datalad/distribution/dataset.py

```diff
@@ -407,7 +408,10 @@ def is_installed(self):
         -------
         bool
         """
-        return self.path is not None and self.repo is not None
+        return \
+            self.path is not None \
+            and self.repo is not None \
+            and exists(self.repo.repo.git_dir)
```

Comment:
Shouldn't then .repo be reset to None if it is physically gone and ATM we afaik can't have a non existing physically on the filesystem repo?

I will merge meanwhile